lib/galaxy/jobs/__init__.py

```diff
@@ -2199,37 +2196,19 @@ def __link_file_check(self):
             # The tool is unavailable, we try to move the outputs.
             return False
 
-    def _change_ownership(self, username, gid):
-        job = self.get_job()
-        external_chown_script = self.get_destination_configuration("external_chown_script", None)
-        if external_chown_script is not None:
-            cmd = shlex.split(external_chown_script)
-            cmd.extend([self.working_directory, username, str(gid)])
-            log.debug('(%s) Changing ownership of working directory with: %s' % (job.id, ' '.join(cmd)))
-            p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            stdout, stderr = p.communicate()
-            stdout, stderr = unicodify(stdout).strip(), unicodify(stderr).strip()
-            if p.returncode != 0:
-                log.error('external script failed.')
-                log.error('stdout was: %s' % stdout)
-                log.error('stderr was: %s' % stderr)
-            assert p.returncode == 0
-
     def change_ownership_for_run(self):
         job = self.get_job()
         external_chown_script = self.get_destination_configuration("external_chown_script", None)
-        if external_chown_script and job.user is not None:
-            try:
-                self._change_ownership(self.user_system_pwent[0], str(self.user_system_pwent[3]))
-            except Exception:
-                log.exception('(%s) Failed to change ownership of %s, making world-writable instead' % (job.id, self.working_directory))
-                os.chmod(self.working_directory, RWXRWXRWX)
+        if job.user is not None:
+            external_chown(self.working_directory, self.user_system_pwent,
+                           external_chown_script, description="working directory")
```

Comment:
Important question: Is it OK not to `os.chmod(self.working_directory, RWXRWXRWX)` in case of error? Or should this be integrated in `external_chown` or just here?

Reply:
Is this change critical for fixing hisitory import/export ? If not I would keep the current behavior and change it in another PR.